plugins/defaultplugins/rpc/services.go

```diff
@@ -76,164 +78,171 @@ type ResyncVppSvc struct {
 	localReqPut := localReq.Put()
 	for _, acl := range request.Acl {
 		localReqPut.ACL(acl)
 	}
 
 	err := localReq.Send().ReceiveReply()
-	return &vppsvc.PutResponse{}, err
+	return &rpc.PutResponse{}, err
 }
 
 // DelACLs deletes one or multiple ACLs by their unique names
 // (forwards the input to the localclient).
-func (svc *ChangeVppSvc) DelACLs(ctx context.Context, request *vppsvc.DelNamesRequest) (*vppsvc.DelResponse, error) {
-	localReq := localclient.DataChangeRequest("vppsvc")
+func (svc *ChangeVppSvc) DelACLs(ctx context.Context, request *rpc.DelNamesRequest) (*rpc.DelResponse, error) {
+	localReq := localclient.DataChangeRequest("rpc")
 	localReqDel := localReq.Delete()
 	for _, aclName := range request.Name {
 		localReqDel.ACL(aclName)
 	}
 
 	err := localReq.Send().ReceiveReply()
-	return &vppsvc.DelResponse{}, err
+	return &rpc.DelResponse{}, err
 }
 
 // PutStaticRoutes creates or updates one or multiple ACLs
 // (forwards the input to the localclient).
 func (svc *ChangeVppSvc) PutStaticRoutes(ctx context.Context, request *l3.StaticRoutes) (
-	*vppsvc.PutResponse, error) {
-	localReq := localclient.DataChangeRequest("vppsvc")
+	*rpc.PutResponse, error) {
+	localReq := localclient.DataChangeRequest("rpc")
 	localReqPut := localReq.Put()
 	for _, route := range request.Route {
 		localReqPut.StaticRoute(route)
 	}
 
 	err := localReq.Send().ReceiveReply()
-	return &vppsvc.PutResponse{}, err
+	return &rpc.PutResponse{}, err
 }
 
 // DelStaticRoutes deletes one or multiple ACLs by their unique names
 // (forwards the input to the localclient).
-func (svc *ChangeVppSvc) DelStaticRoutes(ctx context.Context, request *vppsvc.DelStaticRoutesRequest) (*vppsvc.DelResponse, error) {
-	localReq := localclient.DataChangeRequest("vppsvc")
+func (svc *ChangeVppSvc) DelStaticRoutes(ctx context.Context, request *rpc.DelStaticRoutesRequest) (*rpc.DelResponse, error) {
+	localReq := localclient.DataChangeRequest("rpc")
 	localReqDel := localReq.Delete()
 	for _, route := range request.Route {
 		localReqDel.StaticRoute(route.VRF, route.DstAddr, route.NextHopAddr)
 	}
 
 	err := localReq.Send().ReceiveReply()
-	return &vppsvc.DelResponse{}, err
+	return &rpc.DelResponse{}, err
 }
 
 // ResyncConfig fills data resync request of defaultvppplugin configuration
 // , i.e. forwards the input to the localclient.
-func (svc *ResyncVppSvc) ResyncConfig(ctx context.Context, request *vppsvc.ResyncConfigRequest) (
-	*vppsvc.ResyncConfigResponse, error) {
+func (svc *ResyncVppSvc) ResyncConfig(ctx context.Context, request *rpc.ResyncConfigRequest) (
+	*rpc.ResyncConfigResponse, error) {
 
-	localReq := localclient.DataResyncRequest("vppsvc")
-	for _, intf := range request.Interfaces.Interface {
-		localReq.VppInterface(intf)
-	}
+	localReq := localclient.DataResyncRequest("rpc")
 
-	for _, bd := range request.BDs.BridgeDomains {
-		localReq.BD(bd)
+	if request.Interfaces != nil && len(request.Interfaces.Interface) > 0 {
```

Comment:
Check for length is unnecessary in this case. Applies to if conditions below as well.